src/common/eth-providers/consensus-provider/consensus-provider.service.ts

```diff
@@ -110,74 +119,65 @@ export class ConsensusProviderService {
           return false;
         },
       },
-    );
-
-    return {
-      slotNumber: BigInt(blockHeader.header.message.slot),
-      stateRoot: blockHeader.header.message.state_root,
-      blockRoot: blockHeader.root,
-      parentRoot: blockHeader.header.message.parent_root,
-    };
-  }
-
-  public async getBeaconBlockHeaderOrPreviousIfMissed(slot: bigint): Promise<[ShortBeaconBlockHeader, boolean]> {
-    try {
-      return [await this.getBeaconBlockHeader(slot), false];
-    } catch (e: any) {
+    ).catch((e) => {
       if (404 != e.$httpCode) {
         this.logger.error('Unexpected status code while fetching block header');
         throw e;
       }
-      // if block 64 is missed, try to get next not missed block header
-      const someNotMissedNextBlock = await this.getNextNotMissedBlockHeader(slot + 1n);
+    });
 
-      this.logger.log(
-        `Found next not missed slot [${
-          someNotMissedNextBlock.slotNumber
-        }] root [${someNotMissedNextBlock.blockRoot.toString()}] after slot [${slot}]`,
-      );
+    if (!['finalized', 'head'].includes(String(state))) {
```

Comment:
`head` may be changed over time, but `finalized`? By fork? Anyway, the fetched data has a block header hash or something to identify and can be store, am I right?

Reply:
Now `finalized` is 32, after ~6.5 minutes it will be 64. Exactly like a head that can changes every 12 seconds. I want to make simple service without extracting values of response body. This is caching by ID (slot number, root), not by state

Reply:
I mean `finalized` has a root like `0xsomething` and can be cached by this key.

Reply:
Left TODO

Reply:
Why not to implement? It's super easy.